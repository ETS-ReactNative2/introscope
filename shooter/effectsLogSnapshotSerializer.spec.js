const { newShooterLog } = require('.');
const {
    effectsLogSnapshotSerializer,
} = require('./effectsLogSnapshotSerializer');

global.expect.addSnapshotSerializer(effectsLogSnapshotSerializer);

describe('spySnapshotSerializer', () => {
    const log = newShooterLog();
    log.push(['call', 'function1', [1, 2, 3]]);

    it('naked list', () => {
        expect(log).toMatchSnapshot();
    });

    // it('list of lists', () => {
    //     expect([
    //         newSerializedSpy('A'),
    //         newSerializedSpy('B'),
    //         newSerializedSpy('C'),
    //     ]).toMatchSnapshot();
    // });

    // it('object of lists', () => {
    //     expect({
    //         a: newSerializedSpy('A'),
    //         b: newSerializedSpy('B'),
    //         c: newSerializedSpy('C'),
    //     }).toMatchSnapshot();
    // });

    it('keeps other objects', () => {
        expect(['str', 5, [1, 2, 3], [{ a: 1 }]]).toMatchSnapshot();
    });
});
