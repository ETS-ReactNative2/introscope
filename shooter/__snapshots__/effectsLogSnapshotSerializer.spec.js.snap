// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spySnapshotSerializer keeps other objects 1`] = `
Array [
  "str",
  5,
  Array [
    1,
    2,
    3,
  ],
  Array [
    Object {
      "a": 1,
    },
  ],
]
`;

exports[`spySnapshotSerializer naked list 1`] = `
EffectsLog [
  function1(
      1,
      2,
      3,
  )
]
`;
